Makefile

```diff
@@ -264,6 +264,11 @@ generate-modules: ## Run go mod tidy to ensure modules are up to date
 	go mod tidy
 	cd $(TOOLS_DIR); go mod tidy
 
+.PHONY: download-cluster-api-crd
+download-cluster-api-crd: generate-modules ## Run to download Cluster API CRDs for tests
+	cp -r $(shell go env GOPATH)/pkg/mod/sigs.k8s.io/cluster-api@$(shell go list -f "{{.Version}}" -m sigs.k8s.io/cluster-api)/config/crd/bases/cluster.x-k8s.io_clusters.yaml ./controllers/testdata/crd
```

Comment:
Should we add a CAPI version pin in the Makefile since the CAPI version is already pinned in go.mod?

Reply:
Yes, originally I used this approach (CLUSTER_API_VERSION variable in Makefile), but current approach avoids to keep it and changing every time Clusrer API changing in go.mod

Reply:
Fix it with introducing `CLUSTER_API_VERSION := $(call get_go_version,sigs.k8s.io/cluster-api)` variable. Also tune `get_go_version` a bit without `awk` utilization and native golang format approach.